src/modules/editTask.js

```diff
@@ -0,0 +1,104 @@
+const todoList = document.querySelector('.todo-list');
+
+const runComplete = (inputEl) => {
+  const localData = JSON.parse(localStorage.getItem('tasks'));
+  localData.forEach((task) => {
+    if (inputEl.checked !== false) {
+      if (Number(inputEl.id) === Number(task.index)) {
+        task.completed = true;
+      }
+    } else {
+      task.completed = false;
+    }
+    localStorage.setItem('tasks', JSON.stringify(localData));
+  });
+};
+
+const toggleComplete = (inputElement) => {
+  if (inputElement.checked === false) {
+    inputElement.parentElement.parentElement.classList.remove('strike-thro');
+    inputElement.parentElement.parentElement.parentElement.classList.remove('active');
+  } else {
+    inputElement.parentElement.parentElement.classList.add('strike-thro');
+    inputElement.parentElement.parentElement.parentElement.classList.add('active');
+  }
+};
+
+const showEditInputs = (liElement) => {
+  const editInput = document.querySelector('.edit-input');
+  if (editInput) {
+    editInput.remove();
+  }
+  const input = document.createElement('input');
+  input.type = 'text';
+  input.name = 'editInput';
+  input.value = liElement.textContent;
+  input.classList.add('edit-input');
+  liElement.appendChild(input);
+  input.focus();
+};
+
+function editTask() {
+  todoList.addEventListener('click', () => {
+    const editEl = document.querySelectorAll('.edit-task');
+    editEl.forEach((edit) => {
+      edit.addEventListener('change', (e) => {
+        e.stopPropagation();
+        if (e.target.tagName === 'INPUT' && e.target.type === 'checkbox') {
+          toggleComplete(e.target);
+          runComplete(e.target);
+        }
+      });
+    });
+  });
+}
+
+function editText() {
+  todoList.addEventListener('click', () => {
+    const editEl = document.querySelectorAll('.edit-task');
+    editEl.forEach((edit) => {
+      edit.addEventListener('click', (e) => {
+        e.stopPropagation();
+        if (e.target.tagName === 'LI') {
+          showEditInputs(e.target);
+        }
+      });
+    });
+  });
+}
+
+function updateText() {
+  const editInput = document.querySelector('.edit-input');
+  const localData = JSON.parse(localStorage.getItem('tasks'));
+
+  if (!editInput) {
+    return;
+  }
+  const newText = editInput.value;
+  if (newText !== '') {
+    const liInput = editInput.parentElement;
+    liInput.textContent = newText;
+    localData.forEach((data) => {
+      data.text = newText;
+      localStorage.setItem('tasks', JSON.stringify(localData));
+    });
+  }
+  editInput.remove();
+}
```

Comment:
- [ ] Kindly use the icon for deleting instead of using the three-dot button
- [ ] When you click on the delete button for the first time, all the added tasks get lost until you refresh the page, kindly fix it.
- [ ] When you click on the check box I see the changes in the local storage however you reload the page the changes doesn't reflect on the page too, I am expecting to see the completed checkbox when you reload the page too. 
 
![image](https://user-images.githubusercontent.com/55635977/165976322-f537f7ae-dc6d-4fd2-ba10-ab10eb1f0170.png)